src/LndClient/RPC/Katip.hs

```diff
@@ -228,13 +228,13 @@ catchWalletLock env x = do
   x0 <- x
   case x0 of
     Left LndWalletLocked -> do
-      $(logTM) (newSev env WarningS) "Wallet is locked, needs to be unlocked"
+      $(logTM) (newSev env WarningS) "Wallet possibly is locked, try to unlock"
       unlocked <- lazyUnlockWallet env
       case unlocked of
         Right () -> do
           $(logTM) (newSev env InfoS) "Wallet unlocked"
-          pure x0
-        Left err -> do
+          x
```

Comment:
This is implicit repeat of original RPC call. It's dangerous, some RPCs might be not suitable for safe retries. For example SendCoins. Let's leave it as it was, just returning result whatever is was. Retries are handled separately in other combinator.